From 10459f7cb17a9f9e24813f742919b23f94d95b23 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Wed, 3 Jan 2024 13:58:18 +0800
Subject: [PATCH] Change the Behavior of Name Server 	modified:  
 src/dns_updater.cpp 	modified:   src/ethernet_interface.cpp 	modified:  
 src/ethernet_interface.hpp

---
 src/dns_updater.cpp        |  5 +++++
 src/ethernet_interface.cpp | 15 ++++++++++++---
 src/ethernet_interface.hpp |  6 ++++--
 3 files changed, 21 insertions(+), 5 deletions(-)

diff --git a/src/dns_updater.cpp b/src/dns_updater.cpp
index ea002c4..6447ec2 100644
--- a/src/dns_updater.cpp
+++ b/src/dns_updater.cpp
@@ -569,6 +569,11 @@ int16_t Configuration::setDNSServer(std::string interface, std::vector<std::stri
         return -1;
     }
 
+    if (manager.get().getDHCPConf().dnsEnabled()) {
+        lg2::error("Not support in currnt state.\n");
+        return -1;
+    }
+
     for (auto& ipaddress:servers) {
         try {
             stdplus::fromStr<stdplus::InAnyAddr>(ipaddress);
diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index e710a77..4bfc5a6 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -876,6 +876,11 @@ bool EthernetInterface::nicEnabled(bool value)
 
 ServerList EthernetInterface::staticNameServers(ServerList value)
 {
+    if (manager.get().getDHCPConf().dnsEnabled() && !value.empty()) {
+        lg2::error("Not support in current state: DHCP DNS is Enabled");
+        elog<NotAllowed>(NotAllowedArgument::REASON("Not support in current state"));
+    }
+
     for (auto& ip : value)
     {
         try
@@ -911,8 +916,8 @@ void EthernetInterface::loadNameServers(const config::Parser& config)
 {
     if (manager.get().getDHCPConf().dnsEnabled())
         EthernetInterfaceIntf::nameservers(getNameServerFromResolvd());
-    EthernetInterfaceIntf::staticNameServers(
-        config.map.getValueStrings("Network", "DNS"));
+    else
+        EthernetInterfaceIntf::staticNameServers(config.map.getValueStrings("Network", "DNS"));
 }
 
 void EthernetInterface::loadDomainNames()
-- 
2.34.1


